crates/nu-protocol/src/value/primitive/color.rs

```diff
@@ -0,0 +1,203 @@
+use std::fmt;
+use serde::{Deserialize, Serialize};
+use std::hash::{Hash};
+use std::str::FromStr;
+use bigdecimal::BigDecimal;
+
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Deserialize, Serialize)]
+pub enum ColorData {
+    Rgba(Rgb, u8),
+    Hsla(Hsl, u8),
+    Hsva(Hsv, u8)
```

Comment:
Can we convert from one color format to another? If so, we might want to just pick one format and store in that one, and then offer color conversion commands to get to the other ones.

Reply:
We could, but not all colour space conversions are lossless